pvlib/modelchain.py

```diff
@@ -245,12 +245,15 @@ def get_orientation(strategy, **kwargs):
     return surface_tilt, surface_azimuth
 
 
+# Type for fields that vary between arrays
+_T = TypeVar('T')
+
+
+_PerArray = Union[_T, Tuple[_T, ...]]
```

Comment:
No underscore on either of these. Type variables should be public so other libraries that build on pvlib can inspect them and benefit from them.